util/tls/tls_socket.cc

```diff
@@ -309,35 +322,23 @@ auto TlsSocket::MaybeSendOutput() -> error_code {
 }
 
 auto TlsSocket::HandleRead() -> error_code {
+  if (state_ & READ_IN_PROGRESS) {
```

Comment:
Add a comment along the lines of "Protects against concurrent calls, see MaybeSendOutput()"?